File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int best = 0, sz = s.size(), start;
        for (int i = 0; i < s.size(); i++) {
            int l = i - 1, r = i + 1;
            while (l >= 0 && s[l] == s[i]) l--;
            while (r < sz && s[r] == s[i]) r++;
            while (l >= 0 && r < sz && s[l] == s[r]) { l--; r++; }
            if (r - l - 1 > best) {
                best = r - l - 1;
                start = l + 1;
            }
        }
        return s.substr(start, best);
    }
};

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        result = ""
        
        left = 0
        while left < len(s):
            right = left + 1
            while right < len(s) and s[left] == s[right]:
                right += 1

            if (right - left > len(result)):
                result = s[left:right]

            increase = (len(result) - (right - left)) // 2
            while left - increase >= 0 and right + increase <= len(s):
                check = s[left - increase : right + increase]
                if check == check[::-1] and len(check) > len(result):
                    result = check

                increase += 1

            left += 1

        return result
